### PR TITLE
CKAN gists

### PR DESCRIPTION
**Pull request by sleitner**
This includes changes that allow use of an ACE editor and posting of code snippets as gists to GitHub. In order to build with full functionality (i.e. with the ACE javascript), the related [Ansible PR](https://github.com/cfpb/ckan-installer/pull/13) will need to be merged. 

Should this be its own plugin?

### CODE REVIEW

File: ckanext/cfpb_extrafields/templates/package/resource_read.html [hunk 1]
```diff
@@ -105,8 +110,71 @@
...
             </tbody>
-          </table>
-        </div>
-      </section>
-    {% endif %}
+    </table>
+  </div>
+  <div class="module-content">
+<!-- ============ SHARE ============= -->
+<!-- CODE REVIEW: This must come before the VIEW section below for some reason?! -->
```
Comment (Ooblioob) [inline ckanext/cfpb_extrafields/templates/package/resource_read.html:117]: My only guess here is about the order in which things get loaded.  It must need these resources in order to render, but I don't have any specific details.  Sorry :(

File: ckanext/cfpb_extrafields/fanstatic/post_related_gist.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                                                        this.options,
+                                                        null);
+                        this._snippetReceived = true;
+                    }
+                } 
+            });
+        },
+
+        _postGIST: function(apiurl, basecontent, callback) {
+            var mythis=this; // code review?
```
Comment (Ooblioob) [inline ckanext/cfpb_extrafields/fanstatic/post_related_gist.js:44]: I'm unsure about this too.  It seems like it's unnecessary, you're only using `this` on the next line and in your ajax callback.  In the case of the ajax callback, I don't think it's needed, but I don't have enough experience in this area to say for certain.  I'll reach out to one of our javascript gurus.

**Comment by Ooblioob** [comment]
Review completed.  This looks fine to me, :+1: to merge

**Comment by sleitner** [comment]
Thanks for the review! What do you think about making it a separate extension?

File: ckanext/cfpb_extrafields/fanstatic/post_related_gist.js (hunk 2 above)
Comment (sephcoster) [inline ckanext/cfpb_extrafields/fanstatic/post_related_gist.js:44]: Hey folks. By setting mythis = this at the top of the function you ensure that you're binding the correct context for your $.ajax call further down the page. If you were to pass "this" into the complete function, you'd end up passing a "this" value bound to that call site, and NOT the top level _postGIST "this" value that you intended, so you'd get a totally different value and or "undefined".

Reading through the code again, it looks like you could probably use a "this" when assigning the `.options.gistdesc` value on line 45 without issue, but as soon as you try to call it in a nested function the scope changes unless you .bind() to another scope. Either way I'd stick with it once you've assigned "mythis" if you need to re-use it again so you can keep your head straight reading the function in the future.

Hope that makes sense.  [Here's a good clarification doc if you're interested - that and chapter two are good things to know regarding the peculiarities of "this" in JS.](https://github.com/getify/You-Dont-Know-JS/blob/master/this%20&%20object%20prototypes/ch1.md)

Comment (sleitner) [inline ckanext/cfpb_extrafields/fanstatic/post_related_gist.js:44]: Cool, that makes some sense. And yeah, I need to read up on 'this', so thanks for the reference.

**Comment by Ooblioob** [comment]
I'd be fine if you wanted to make it a separate extension.  It seems like that would be a cleaner approach.  I can merge it now and we can extract it later if you'd like.

**Pull request merged**